### PR TITLE
[s] Adds the top secret room to centcom

### PR DESCRIPTION
**Pull request by Incoming5643**
The final destination of players who manage to cosign themselves to nullspace. It will allow them to travel back to the station after touching the tile at the top of the room there.

ghost copy

Said room is an homage to a room that served a similar purpose in the legend of zelda.

Also, memes.

### CODE REVIEW

File: code/modules/awaymissions/super_secret_room.dm [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			never_ends.name = "Overextending The Joke: First Place"
+			never_ends.desc = "And so we are left alone with our regrets."
+		else
+			y += 2
+
+	speaking = FALSE
+	times_spoken_to++
+
+/obj/structure/speaking_tile/proc/SpeakPeace(list/statements)
+	for(var/i=1, i<=statements.len, i++)
```
Comment (RemieRichards) [inline code/modules/awaymissions/super_secret_room.dm]: ```for(var/i in 1 to statements.len)```

**Comment by PKPenguin321** [comment]
cute reference and also pretty eerie
nice way of handling errors as well
:+1:

File: code/modules/mob/living/life.dm [hunk 2]
```diff
@@ -8,7 +8,12 @@
 	if (notransform)
 		return
 	if(!loc)
-		return
+		if(client)
+			for(var/obj/effect/landmark/error/E in landmarks_list)
+				loc = E.loc
```
Comment (Jordie0608) [inline code/modules/mob/living/life.dm]: Could you add an admin log and message that this happened?

File: code/modules/awaymissions/super_secret_room.dm [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	verb_say = "intones"
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "speaking_tile"
+	layer = 5
+	var/speaking = FALSE
+	var/times_spoken_to = 0
+	var/list/shenanigans = list()
+
+/obj/structure/speaking_tile/New()
+	var/savefile/S = new /savefile("data/npc_saves/Poly.sav")
```
Comment (Iamgoofball) [inline code/modules/awaymissions/super_secret_room.dm]: uh

Comment (Incoming5643) [inline code/modules/awaymissions/super_secret_room.dm]: shhh... it's lore guys.

lore.

File: code/modules/awaymissions/super_secret_room.dm [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+			if(times_spoken_to * 0.1 == round(times_spoken_to * 0.1))
+				SpeakPeace(list("That's [times_spoken_to] times you've spoken to me by the way."))
+		if(15)
+			SpeakPeace(list("See? Isn't this fun?","Now you can mash this for hours without getting bored.","Anyway I'll leave you it."))
+		if(51)
+			SpeakPeace(list("The fun never ends around here.", "The Poly text files stores up to 500 statements.", "But you've probably heard a few repeats by now."))
+		if(100)
+			SpeakPeace(list("And that's a solid hundred.", "Good hustle I guess.", "You've probably heard a lot of repeats by now."))
+		if(101)
+			SpeakPeace(list("I hope your getting the reference this room is presenting.", "As well as the more obscure meta reference this conversation is presenting.", "This stuff has layers."))
```
Comment (Iamgoofball) [inline code/modules/awaymissions/super_secret_room.dm]: >>>your

maxresdefault 1
triggered
maxresdefault
o1v6bfvj_200x200
4a3c3e9ddb89babe1ae32689dd958171

**Comment by xxalpha** [comment]
You went to nullspace in zelda?

**Comment by Iamgoofball** [comment]
@xxalpha in some zelda games, failed level loads would drop you in a secret room with some rupees as an apology and as error handling so you didn't load garbage levels

File: code/modules/awaymissions/super_secret_room.dm [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+/obj/structure/speaking_tile/proc/SpeakPeace(list/statements)
+	for(var/i=1, i<=statements.len, i++)
+		say("<span class='deadsay'>[statements[i]]</span>")
+		if(i != statements.len)
+			sleep(30)
+
+/obj/item/rupee
+	name = "weird crystal"
+	desc = "Your excitement boils away as you realize it's just colored glass. Why would someone horde these things?"
```
Comment (xxalpha) [inline code/modules/awaymissions/super_secret_room.dm]: hoard

Comment (Incoming5643) [inline code/modules/awaymissions/super_secret_room.dm]: FOR THE HOARD

**Comment by Incoming5643** [comment]
Specifically it's the Chris Houlihan Room From a link to the past.

File: code/modules/mob/living/life.dm (hunk 2 above)
Comment (Incoming5643) [inline code/modules/mob/living/life.dm]: I have, below this line.

**Comment by Incoming5643** [comment]
All issues addressed

File: code/modules/mob/living/life.dm [hunk 6]
```diff
@@ -8,7 +8,14 @@
...
+				break
+			message_admins("[key_name(src)] was found to have no .loc with an attached client, if the cause is unknown it would be wise to ask how this was accomplished.")
```
Comment (Jordie0608) [inline code/modules/mob/living/life.dm]: `key_name_admin()` for `message_admins()`

**Comment by X-TheDark** [comment]
Holy shit, this is really creative and awesome.

**Comment by Saegrimr** [comment]
▲
▲  ▲

**Pull request merged**